jupyterlab_caip_optimizer/src/types.ts

```diff
@@ -6,41 +6,54 @@ export interface AsyncState<T> {
   data?: T;
 }
 
-// Optimizer API
 export interface MetricSpec {
   goal: GoalType;
   metric: string;
 }
 
-export enum GoalType {
-  GOAL_TYPE_UNSPECIFIED = 'GOAL_TYPE_UNSPECIFIED',
-  MAXIMIZE = 'MAXIMIZE',
-  MINIMIZE = 'MINIMIZE',
-}
+/**
+ * 'Const assignment" used to control the string inputs in dropdown input fields.
+ * String enums don't support reverse mapping, and plain string literal types aren't iterable.
+ */
+export const GoalTypeList = [
+  'MAXIMIZE',
+  'MINIMIZE',
+  'GOAL_TYPE_UNSPECIFIED',
+] as const;
```

Comment:
Is the `as const` necessary? That's surprising to me

Reply:
If there isn't `as const` here, the `typeof` in the next line will widen the types to `string`. `As const` makes sure that the literal types don't get widened so that when we try to extract types from the array, the type will be `'MAXIMIZE' | 'MINIMIZE' | 'GOAL_TYPE_UNSPECIFIED'` instead of `string`.
Reference: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions